src/Prometheus.Client.HttpClient/MessageHandlers/HttpClientInProgressHandler.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Prometheus.Client.HttpClient.MessageHandlers
+{
+    public sealed class HttpClientInProgressHandler : HttpClientMessageHandlerBase<IMetricFamily<IGauge, ValueTuple<string, string, string, string>>, IGauge>
+    {
+        public HttpClientInProgressHandler(IMetricFactory metricFactory, string clientName)
+            : base(metricFactory, clientName)
+        {
+        }
+
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            HttpResponseMessage response = null;
+            var metric = CreateMetric(request, null);
```

Comment:
Looks like `status_code` label always null. Let's delete this label from InProgress metric.

Reply:
@phnx47 
In current handlers structure, I have to keep Tuple with 4 strings including status_code.

Reply:
Tuple with 3 strings in InProgress
Tuple with 4 strings in RequestDuration

Reply:
If you would see in HandlerBase, it's in class signature to have 4 string tuples. I would need to tear down the current implementation to keep 3 strings tuple in InProgress.

Reply:
Yes, it is why I suggested to combine handlers before. Base class useless here.

```c#
public class HttpClientMetricsHandler : DelegatingHandler
{
    private readonly string _clientName;

    private readonly IMetricFamily<IGauge, (string, string, string)> _inProgress;
    private readonly IMetricFamily<IHistogram, (string, string, string, string)> _requestDuration;

    public HttpClientMetricsHandler(ICollectorRegistry registry, string clientName)
    {
        _clientName = clientName;
        var metricFactory = new MetricFactory(registry);

        _inProgress = metricFactory.CreateGauge(Defaults.InProgresMetricName, Defaults.InProgresMetricHelp, ("method", "host", "client"));
        _requestDuration = metricFactory.CreateHistogram(Defaults.RequestDurationMetricName, Defaults.RequestDurationMetricHelp, ("method", "host", "client", "status_code"));
    }

    protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
    {
        var stopWatch = Stopwatch.StartNew();
        HttpResponseMessage response = null;
        var method = request?.Method.Method;
        var host = request?.RequestUri?.Host;
        try
        {
            _inProgress.WithLabels((method, host, _clientName)).Inc();
            response = await base.SendAsync(request, cancellationToken);
            return response;
        }
        finally
        {
            _inProgress.WithLabels((method, host, _clientName)).Dec();
            stopWatch.Stop();
            _requestDuration.WithLabels((method, host, _clientName, response?.StatusCode.ToString())).Observe(stopWatch.Elapsed.TotalSeconds);
        }
    }
}
```

Reply:
You still can use as separate handlers but must not be any empty label value.

Reply:
Yeah I can use separate handlers by keeping `metric` instance in derived handler classes. 

> But must not be any empty label value.

Let me refactor the implementation then.